framework/src/base/FEProblemBase.C

```diff
@@ -200,6 +200,55 @@ FEProblemBase::FEProblemBase(const InputParameters & parameters) :
   _eq.parameters.set<FEProblemBase *>("_fe_problem_base") = this;
 }
 
+void FEProblemBase::newAssemblyArray(NonlinearSystemBase & nl)
+{
+  unsigned int n_threads = libMesh::n_threads();
+
+  _assembly.resize(n_threads);
+  for (unsigned int i = 0; i < n_threads; ++i)
+    _assembly[i] = new Assembly(nl, couplingMatrix(), i);
```

Comment:
I will take a look, and this code has been in moose for a while and I am not sure why the original developer does not use a smart pointer.